.cirrus.yml

```diff
@@ -0,0 +1,559 @@
+freebsd_instance:
+  cpu: 1
+  memory: "1"
+
+install_template: &INSTALL_PLUGIN
+  requirements_script: pkg install -y jq
+  install_script: |
+    release=$(jq -r '.release' $PLUGIN_FILE)
+    name=$(jq '.name' $PLUGIN_FILE)
+    packagesite=$(jq '.packagesite' $PLUGIN_FILE)
+    fingerprints=$(jq -r '.fingerprints | keys[]' $PLUGIN_FILE)
+    plugin_repo=$(jq -r '.artifact' $PLUGIN_FILE)
+    pkgs=$(jq -r '.pkgs | join(" ")' $PLUGIN_FILE)
+
+    pkg_dir=/usr/local/test
+    repos_dir="${pkg_dir}/repos"
+    fingerprints_dir="${pkg_dir}/fingerprints"
+
+    echo "Creating main repos dir: ${repos_dir}"
+    mkdir -p $repos_dir
+    export REPOS_DIR=$repos_dir
+
+    pkg_conf_path="${repos_dir}/test.conf"
+    echo "iocage-plugins: {" > $pkg_conf_path
+    echo "url: $packagesite," >> $pkg_conf_path
+    echo "signature_type: \"fingerprints\"," >> $pkg_conf_path
+    echo "fingerprints \"${fingerprints_dir}\"," >> $pkg_conf_path
+    echo "enabled: true" >> $pkg_conf_path
+    echo } >> $pkg_conf_path
+    echo "Created test pkg config file:"
+    cat $pkg_conf_path
+
+    trusted_fingerprints="$fingerprints_dir/trusted"
+    mkdir -p "${trusted_fingerprints}"
+
+    for repo_name in $fingerprints
+    do
+      repo_fingerprints=$(jq -rc '."fingerprints"."'${repo_name}'"[]' $PLUGIN_FILE)
+
+      repo_count=1
+      echo $repo_fingerprints | while IFS='' read f
+      do
+        echo "Creating fingerprint file for repo:"
+        echo $f
+
+        function=$(echo $f | jq -r '.function')
+        fingerprint=$(echo $f | jq -r '.fingerprint')
+        file_path=${trusted_fingerprints}/${repo_name}_${repo_count}
+
+        echo "Creating new fingerprint file: ${file_path}"
+
+        echo "function: $function" > ${file_path}
+        echo "fingerprint: $fingerprint" >> ${file_path}
+
+        repo_count=$(expr $repo_count + 1)
+      done
+    done
+
+    if [ "$(which git)" = "" ]
+    then
+      pkg install -y git-lite || pkg install -y git
+    fi
+
+    release_branch="$(freebsd-version | cut -d '-' -f1)-RELEASE"
+    git clone -b ${release_branch} ${plugin_repo} ./plugin || git clone -b master ${plugin_repo} ./plugin
+
+    # Cleanup before plugin pkg installation
+    if ! echo ${pkgs} | grep -q "git"
+    then
+      pkg remove -y git-lite git jq
+      pkg autoremove -y
+    fi
+
+    echo "Test install plugin pkgs for plugin: $name"
+    echo "Installing $name pkgs: $pkgs"
+    pkg install -y $pkgs
+
+    if [ -d "./plugin/overlay" ]
+    then
+      echo "Found overlay folder"
+      cp -r ./plugin/overlay/ /
+    fi
+
+    ./plugin/post_install.sh
+
+    if [ -f ./plugin/pre_update.sh ]
+    then
+      ./plugin/pre_update.sh
```

Comment:
I found another corner case in the home assistant plugin, that shows a [false negative](https://github.com/tprelog/test/runs/1957637076) --- If you look before the error, you can see the service has started and status shows it's running -- ( what I'm doing there is OT, but I have another solution for that in the zwavejs2mqtt plugin )

On the other hand - maybe this is not really _false_ negative because it shows a user can not expect to run the upgrade after the fact -- I mean, is it reasonable to think if a plugin can not run the update scripts on a clean install, it is not going to work when the user tries to update the plugin after later on?

Even if we do not run the scripts, I'm fine with that -- Should we still check if they are at least executable when they do exists?

Reply:
Yes we should / can still check if they are executable :)

Reply:
During the first implementation of these tests I was thinking it would be a good idea (like @tprelog commented) to test if the users would be able to upgrade the plugins just after a clean installation (that's the reason behind these executions). On the other hand, it is maybe not a probable test case after all, I'm guessing the majority of users upgrade a plugin mostly because of a FreeBSD version upgrade.

Have no problem removing this for now and only check if the scripts are executable (I know I've personally forgotten to change the mode of these, and the post_install, scripts in my plugins), should be changed now.

Reply:
As these scripts are no longer executed I will resolve this. Feel free to open up or add a new comment if something is still unclear/feels wrong.